WIP - Add setting to keep suppresion test open until the end of the window

denar50 opened the pull request:
## Summary

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

denar50 committed:
add an advanced setting to control the behavior of the suppression window on alert closure

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

shahzad31 committed:
[Synthetics] Pass readonly to monitor spaces component !! (#230714)

## Summary

Pass readonly to monitor spaces component, caught this while checking
project monitor config, where it should have been disabled !!

### Before
<img width="1445" height="402" alt="image"
src="https://github.com/user-attachments/assets/aed1f3b3-e802-4468-9685-b0960d4e54ca"
/>



### After
<img width="1388" height="562" alt="image"
src="https://github.com/user-attachments/assets/a876b33e-61d4-48e0-a0dd-fbb8352be4de"
/>

jedrazb committed:
fix: cutom esql tool runtime errors (#230745)

## Summary

- Make esql tools always return `ToolResultType.tabularData`
- Use es cli for building esql query 
- Actually there is a bug with named params, reported here
https://github.com/elastic/elasticsearch-specification/issues/5083

What I'm not doing?
- Figuring out how to wire ToolResultType everywhere ... the tool result
type can still be `any`. This just fixes runtime errors as we are
returning right `type` in the response

### Screenshot

Built in and custom esql tools work in agent runtime:


<img width="682" height="737" alt="Screenshot 2025-08-06 at 12 13 29"
src="https://github.com/user-attachments/assets/9523d82f-58aa-4eab-b653-e8f97f15115e"
/>

PhilippeOberti committed:
[Security Solution] unskip alerts_charts Cypress test (#230443)

## Summary

Test is passing locally. Unskipping and running Flaky test runner to
verify.

Flaky Test Runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9012
All tests passed!

https://github.com/elastic/kibana/issues/192459

delanni committed:
[ci] .gitignore Moon's caches (#230795)

## Summary
prepare for moon: ignore .moon/cache to prevent not staged files when
switching working branches

PhilippeOberti committed:
[Security Solution] unskip alert_status Jest test (#230448)

## Summary

Unskipping as test is passing locally without taking much time.

https://github.com/elastic/kibana/issues/216474

PhilippeOberti committed:
[Security Solution] unskip mitre_attack Jest test (#230712)

## Summary

Unskipping as test is passing locally without taking much time.

https://github.com/elastic/kibana/issues/176002

PhilippeOberti committed:
[Security Solution] unskip alert_details_left_panel_prevalence_tab Cypress test (#230711)

## Summary

Test is passing locally. Unskipping and running Flaky test runner to
verify.

Flaky Test Runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9039
All tests passed!

https://github.com/elastic/kibana/issues/229360
https://github.com/elastic/kibana/issues/229361

PhilippeOberti committed:
[Security Solution] unskip timelines export Cypress test (#230707)

## Summary

Test is passing locally. Unskipping and running Flaky test runner to
verify.

Flaky Test Runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9038
All tests passed!

https://github.com/elastic/kibana/issues/187550

PhilippeOberti committed:
[Security Solution] unskip alert_table_controls Cypress test (#230705)

## Summary

Test is passing locally. Unskipping and running Flaky test runner to
verify.

Flaky Test Runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9037
All tests passed!

https://github.com/elastic/kibana/issues/207117

cesco-f committed:
[Synthetics] Renamed ping to latestPing (#230756)

Just a refactor to rename the property `ping` to `latestPing` in the
`AlertStatusMetadata`.

PhilippeOberti committed:
[Security Solution] unskip timeline template creation Cypress test (#230501)

## Summary

Test was failing locally because when creating the timeline template, we
were expecting to have some results but there aren't any documents for
the default `Today` date. Skipping the check until we have selected the
proper timeframe.

Flaky Test Runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9019
All tests passed!

https://github.com/elastic/kibana/issues/183579

PhilippeOberti committed:
[Security Solution] unskip alert_details_preview_panel_rule_preview Cypress test (#230753)

## Summary

Test is passing locally. Unskipping and running Flaky test runner to
verify.

Flaky Test Runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9041
All tests passed!

https://github.com/elastic/kibana/issues/229398

dgieselaar committed:
[OpenTelemetry] Tracing utils (#228448)

Refactor the inference-specific span utilities into more general helper
functions, and make sure spans are appropriately parented.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

juliaElastic committed:
[Fleet] fix output ssl config order (#230758)

## Summary

While testing, encountered another issue with ssl output config. 
It seems that the preconfig comes with some defaults (`{ certificate:
'', certificate_authorities: [] }`) that override the config set in
advanced yaml on the UI.
Change the order so that the advanced yaml comes after preconfig.

To verify:
- Add an elasticsearch output with Advanced YAML config:
```
ssl:
  certificate_authorities:
    - /test/ca.crt
  verification_mode: "none" 
```
- Add output to an agent policy
- View full policy JSON and verify the ssl config is correct

<img width="818" height="1169" alt="image"
src="https://github.com/user-attachments/assets/2c654805-b1f3-4364-a138-fc4d6bc22c52"
/>
<img width="1512" height="1239" alt="image"
src="https://github.com/user-attachments/assets/e2849f17-83d3-4d87-90f7-4b68b817143e"
/>

- Preconfiguration should still work
```
xpack.fleet.outputs:
  - hosts:
    - https://localhost:9200
    id: eck-fleet-agent-output-elasticsearch
    name: eck-elasticsearch
    ssl:
      certificate_authorities:
      - /test/ca.crt
    type: elasticsearch

xpack.fleet.agentPolicies:
  - name: Test preconfigured
    id: test-preconfigured
    is_managed: false
    namespace: default
    monitoring_enabled: []
    package_policies: []
    data_output_id: eck-fleet-agent-output-elasticsearch
```

<img width="1527" height="1236" alt="image"
src="https://github.com/user-attachments/assets/b74aa02e-ac79-44f0-81a4-d3788608dd27"
/>


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

denar50 committed:
fix copy

denar50 commented:
/ci

crespocarlos committed:
[Infra] Remove custom search bar in favor of the unified search bar (#230009)

## Summary


This PR replaces the custom search bar used across Infra with the
Unified Search bar. The custom search bar had some usability problems,
as demonstrated below


![custom_search_bar](https://github.com/user-attachments/assets/f4bba46e-94ab-4ea1-b477-83afe926dc63)


### Tests

![inventory_searchbar](https://github.com/user-attachments/assets/aa741358-47c3-4496-af89-4ab285c6370e)


![metrics_explorer_searchbar](https://github.com/user-attachments/assets/19f4e9d3-7c99-4047-ab49-37845613c13d)


![inventory_rule_flyout_searchbar](https://github.com/user-attachments/assets/b53672f9-8481-4926-b0fe-4b79a264e13c)


**Alert UIs**

![inventory_rule_flyout_searchbar](https://github.com/user-attachments/assets/85368ccb-69e1-4c68-bd27-af28f6814c96)


![metric_threshold_search_bar](https://github.com/user-attachments/assets/63894e38-de53-465e-a473-4de1a9ebd687)


![metric_threshold_custom_equation_searchbar](https://github.com/user-attachments/assets/b761b700-0ab8-434e-9d28-48b706f46ab6)

pheyos committed:
FTR - fix esArchiver duplicate doc ingestion (#229457)

## Summary

This PR fixes duplicate document creation in esArchiver by generating an
`_id` for index (non-data-stream, non-time-series) documents that don't
have an id already.

### Details

- Under some circumstances, the `es-helper-bulk` that is used by
esArchiver can ingest a duplicate document (just with different id), see
investigations [here](https://github.com/elastic/kibana/pull/228556) and
[here](https://github.com/elastic/kibana/pull/223043), also bug report
[here](https://github.com/elastic/elasticsearch-js/issues/2924).
- With explicitly setting the id, the flakiness didn't show up anymore,
which matches the expected behavior as of the [bulk
docs](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-bulk)
`A create action fails if a document with the same ID already exists in
the target An index action adds or replaces a document as necessary.`
- In order to unblock testing, this PR is actually working around the
underlying problem, which should still be investigated separately

---------

Co-authored-by: Dzmitry Lemechko <dzmitry.lemechko@elastic.co>

mattkime committed:
Shared reindexing service / plugin (#228705)

Part of https://github.com/elastic/kibana/issues/218984 and
https://github.com/elastic/kibana/issues/230023

## Summary

Initial creation of ReindexService - extraction of Upgrade Assistant's
reindexing functionality into a separate service. Future PRs will
provide a meaningful API, rather than just whats needed to maintain
current functionality.

Summary of actions
- Create ReindexService plugin
- Create UpgradeAssistant server/common packages to share code between
UpgradeAssistant and ReindexService
- UpgradeAssistant used an imported singleton class Version to perform
version related calculations - now passing the instance as a function
argument.

Testing this code manually takes a number of steps since its disabled in
`main` and will be until the next major version.

How to test - 
1. add `xpack.upgrade_assistant.ui.enabled: true` to
`config/kibana.dev.yml` to enable Upgrade Assistant
2. Because ES is in charge of providing index deprecation info, we'll
mock it so Kibana thinks we need to reindex our sample indices and data
streams. Mock as shown here -
https://github.com/elastic/kibana/pull/230021 - note that some of the
data stream index references will need to be updated
3. Start kibana
4. Add sample data 
5. go to upgrade assistant and try resolving
`kibana_sample_data_ecommerce` , `kibana_sample_data_logs`, and
`kibana_sample_data_flights`
6. Verify that the indices and data streams have been reindexed.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Ignacio Rivas <rivasign@gmail.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

dokmic committed:
[Core] Dependency Injection (#184932)

## Summary

### Motivation
- *Decoupling* should improve reusability, testability, and maintainability.
- Prevent *circular dependencies*.
- *Inversion of Control* should help to get rid of dependencies forwarding and build clear abstraction layers.
- *Clean Code* should provide declarativity and minimize the amount of boilerplate code.

### Why InversifyJS?
1. Familiar and straightforward API. (decorators-based API)
2. Large community. (~30k projects & ~1.2M weekly downloads)
3. Regularly maintained.
4. A kind of standard for DI in JS.
5. Framework agnostic. (works with vanilla/React/Angular/Vue)
6. Support of async dependencies.
7. Support of hierarchical DI.

### Documentation
The developer documentation is in the related `README.md`.

### Examples
This PR contains an example plugin that can be used as a starting point.

jennypavlova committed:
[o11y] feedback buttons redesign (#230413)

Closes #229881 

## Summary
This PR changes the feedback button position, text, and design. It
removes the unused code I found related to the buttons. The text changes
from "Tell us what you think!" to "Give feedback" and "Tell us what you
think (K8s)!" to "Give feedback (K8s)". It should be part of the header
links menu - the only exception will be the ML feedback button in the
flyout in infra and host UIs, because they already have the button (the
design still changes), for example:

<img width="2298" height="1920" alt="image"
src="https://github.com/user-attachments/assets/8016f527-1fb0-456e-b934-d97ec8ea7c0d"
/>

- SLO

| Before | After |
| ------ | ------ | 
| <img width="3810" height="1022" alt="image"
src="https://github.com/user-attachments/assets/3c95b5a9-a9ba-4c27-adaf-90ce441319b3"
/> | <img width="3304" height="1040" alt="image"
src="https://github.com/user-attachments/assets/d9dd0bde-cad9-43d6-bb04-b62ccb1f00b5"
/> |

- APM 


https://github.com/user-attachments/assets/57f3eefc-7e46-40f3-af1a-6bfc1c433df6

- Metrics / Infra


https://github.com/user-attachments/assets/92b4a8a2-4717-42ec-8bc6-5bd801eacc43

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

crespocarlos committed:
[APM] Remove service map v1 (#230364)

closes [#218911](https://github.com/elastic/kibana/issues/218911)

## Summary

Remove all the code related to the service map powered by scripted
metrics aggregation in favor of the new approach.


### How to test

- Navigate to APM > Services Inventory > Service map
- It should just work

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Copilot committed:
Add simple retry logic to CSP plugin getInstallation call with proper error handling (#229804)

Fixes 
- #182268

The Cloud Security Posture plugin's `getInstallation` call during setup
could fail permanently on transient Elasticsearch connection issues
without any retry mechanism. This change adds simple retry logic to
handle temporary ES connection failures and ensures proper error
handling when all retries are exhausted.

## Changes Made

- **Added retry logic** using `p-retry` (already available in the
project) around the `getInstallation` call
- **Configured simple retry** with 3 retries and no delays between
attempts
- **Added proper error logging** for failed retry attempts with
remaining retry count
- **Improved error propagation** - errors are now properly logged when
all retries are exhausted instead of being silently swallowed
- **Follows established patterns** from other plugins that use similar
retry logic

## Implementation Details

The retry configuration includes:
- 3 retry attempts (4 total attempts including initial)
- No delays between retries for immediate retry behavior
- Warning logs on each failed attempt showing remaining retries and
error details
- Proper error logging when all retries are exhausted with clear
messaging

## Testing

Added comprehensive test cases covering:
- Successful retry after initial failures
- Complete failure handling when all retries are exhausted
- Verification that final errors are properly logged for debugging

The solution provides resilience to the specific operation that was
identified as problematic while maintaining all existing functionality
and ensuring proper error visibility for debugging.

<!-- START COPILOT CODING AGENT TIPS -->
---

💬 Share your feedback on Copilot coding agent for the chance to win a
$200 gift card! Click
[here](https://survey.alchemer.com/s3/8343779/Copilot-Coding-agent) to
start the survey.

---------

Co-authored-by: Maxim Kholod <maxim.kholod@elastic.co>
Co-authored-by: maxcold <478762+maxcold@users.noreply.github.com>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

olapawlus committed:
[Examples plugin] Deprecated uiAction methods – Search panel rewrite (#230400)

Solves partially https://github.com/elastic/kibana/issues/227743

viduni94 committed:
[Obs AI Assistant] Expand the evaluation suite for ES|QL generation (#228476)

Closes https://github.com/elastic/obs-ai-assistant-team/issues/316

## Summary

This PR adds new ES|QL scenarios to both current and new eval
frameworks.

## Scores on the `main` branch

### GPT 4o

```
-------------------------------------------
Model azure-gpt4o scored 119.2 out of 189
-------------------------------------------
-------------------------------------------
Model azure-gpt4o Scores per Category
-------------------------
Category: ES|QL query generation - Scored 119.2 out of 189
-------------------------------------------
```

<img width="641" height="482" alt="image"
src="https://github.com/user-attachments/assets/22ad981b-88c3-486a-8e2e-d232133daa7d"
/>

### GPT 4.1

```
-------------------------------------------
Model azure-gpt41 scored 136 out of 189
-------------------------------------------
-------------------------------------------
Model azure-gpt41 Scores per Category
-------------------------
Category: ES|QL query generation - Scored 136 out of 189
-------------------------------------------
```

<img width="643" height="484" alt="image"
src="https://github.com/user-attachments/assets/efc1ab96-9c2d-4a45-b59a-81fe8ac5e5f4"
/>

### Claude 3.5

```
-------------------------------------------
Model bedrock-sonnet-35 scored 141.5 out of 188
-------------------------------------------
-------------------------------------------
Model bedrock-sonnet-35 Scores per Category
-------------------------
Category: ES|QL query generation - Scored 141.5 out of 188
-------------------------------------------
```

<img width="649" height="487" alt="image"
src="https://github.com/user-attachments/assets/2207d8ff-6bc4-425c-a691-8e409d576b42"
/>

### Claude 3.7

```
-------------------------------------------
Model bedrock-sonnet-37 scored 160 out of 188
-------------------------------------------
-------------------------------------------
Model bedrock-sonnet-37 Scores per Category
-------------------------
Category: ES|QL query generation - Scored 160 out of 188
-------------------------------------------
```

<img width="655" height="486" alt="image"
src="https://github.com/user-attachments/assets/cb66b413-487d-48ac-8afb-0a4a6b4a17c5"
/>


### Gemini 2.0 Flash

```
-------------------------------------------
Model gemini-2-flash scored 66 out of 189
-------------------------------------------
-------------------------------------------
Model gemini-2-flash Scores per Category
-------------------------
Category: ES|QL query generation - Scored 66 out of 189
-------------------------------------------
```

<img width="580" height="481" alt="image"
src="https://github.com/user-attachments/assets/f94b82d0-616d-4302-9500-0f4262f291e9"
/>

### Gemini 2.5 Flash

<img width="643" height="482" alt="image"
src="https://github.com/user-attachments/assets/91793f12-2535-4f38-ba0c-39e0f34d7df4"
/>

### Gemini 2.5 Pro

<img width="608" height="386" alt="image"
src="https://github.com/user-attachments/assets/9c5759df-826a-47e9-800a-9ec59fe8d496"
/>



Note:
- Gemini 2.5 Pro was used to generate some of the ES|QL and then
iterated on it.

### Checklist

- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

dkirchan committed:
[Security Solution] [SKA] Moved cypress folder  (#230198)

Summary

Part of https://github.com/elastic/kibana-team/issues/1503

This PR is about moving security solution cypress tests from
`x-pack/test/security_solution/security_solution_cypress` to
`x-pack/solutions/security/test/security_solution_cypress`

Before

x-pack/test/security_solution/security_solution_cypress

After:

x-pack/solutions/security/test/security_solution_cypress

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Dzmitry Lemechko <dzmitry.lemechko@elastic.co>

kapral18 committed:
[Index Management] Fix filter list button styling (#230828)

Addresses #220645

- Remove redundant border properties from
`.componentTemplates__filterListButton`
- Align color in ::after element with main the rest of filterListButton
and searchBox

-----------------------------------------

Before:

<img width="1257" height="933" alt="SCR-20250806-qvhp"
src="https://github.com/user-attachments/assets/ad270a62-6696-42d9-be0c-895b22036baf"
/>


After:

<img width="1245" height="915" alt="SCR-20250806-qlsq"
src="https://github.com/user-attachments/assets/5f03a750-41a1-425c-a2fa-9542b7535d89"
/>

maximpn committed:
[Security Solution] Create mock prebuilt rules Fleet packages for integration testing programmatically  (#230409)

## Summary

This PR improves control and transparency over mocking prebuilt rule packages (`security_detection_engine` and `endpoint`) for integration testing purposes.

## Details

Failed and flaky integration tests troubleshooting and fixing is hindered by using pre-zipped Fleet packages. Figuring out what prebuilt rules are zipped there requires unzipping the archives. Additionally making any changes requires zipping a new version of the package.

Programmatic package creation unveils much more focused, flexible and transparent prebuilt rules mocking. Thanks to that approach it's possible to get rid of existing mock package used for integration testing and replace them with utility functions.

## Flaky test runner

- ✅ `ess_air_gapped_with_bundled_large_package.config.ts` [200 runs](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9003) [200 runs](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9044)
- ✅ `ess_air_gapped_with_bundled_packages.config.ts` [200 runs](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9004) [200 runs](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9045)

SrdjanLL committed:
[HF Data Loader] Load onechat datasets with externally defined mappings.  (#230252)

## Summary
Closes: https://github.com/elastic/kibana/issues/227993

This PR introduces OneChat datasets to the HuggingFace dataset loader
CLI tool.

### Key Changes

* OneChat Data Loading Support
- Added support for loading OneChat datasets (CSV-based) from
the `elastic/OneChatAgent` repository.
- OneChat datasets can now be specified using
the `onechat/<directory>/<dataset>` syntax.
* Refactored File Download and Parsing
    -   Extracted HuggingFace file download logic for reuse.
- Separated CSV and JSON parsing into dedicated functions for clarity
and maintainability. Previously only JSON download was supported. This
uses `papaparse` library for csv parsing which was very handy for
parsing multi-line csv files.
* Configurable Kibana URL
- The loader now accepts a `--kibana-url` flag to override
auto-discovery of the Kibana instance. This is needed locally for the
evaluation framework that uses scout server running on port `5620`.
* Documentation Updates:
- Expanded the README with clear instructions for loading both
HuggingFace and OneChat datasets.
- Added OneChat eval specific instructions in the corresponding README
* Fixed a bug with the existing dataset loader that was previously only
storing one document in ES.

### How to Test

Follow the instructions from the updated README:

1.  Prerequisites

    -   Ensure you have a running Kibana and Elasticsearch instance.
    -   Obtain a HuggingFace access token. 
- (For OneChat datasets) Ensure your HuggingFace account is part of the
Elastic organization.
2.  Run the Loader
```bash
HUGGING_FACE_ACCESS_TOKEN=<YOUR_TOKEN> \
node --require ./src/setup_node_env/index.js \
  x-pack/platform/packages/shared/kbn-ai-tools-cli/scripts/hf_dataset_loader.ts \
  --datasets onechat/knowledge-base/users,huffpost \
  --limit 100 \
  --clear \
  --kibana-url http://<username>:<password>@localhost:5601
```

3.  Verify

    -   Check the logs for successful ingestion and embedding.
    -   Confirm the new indices in your local ES instance.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)


---------

Co-authored-by: Quynh Nguyen (Quinn) <43350163+qn895@users.noreply.github.com>

legrego committed:
Remove set-value dependency (#230295)

## Attention Code Owners!

Please test these changes prior to approving. You know your domain MUCH
better than I do, and I need your help to ensure this does not introduce
a regression. Thank you!

----

This pull request removes the `set-value` dependency from the project
and replaces its usage with the safer and more secure
`@kbn/safer-lodash-set` package. Additionally, it updates related
configuration files to reflect this change.

### Dependency removal and replacement:

*
[`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L1363):
Removed the `set-value` dependency.
*
[`src/platform/plugins/shared/vis_types/timeseries/server/lib/vis_data/helpers/overwrite.ts`](diffhunk://#diff-c1aefe228bdb6aa06e0d8722a191112b3fd2ff1f11f4b80c5529726bd39246c2L10-R10):
Replaced `set-value` with `@kbn/safer-lodash-set` for setting object
paths.
*
[`x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/ml/bulk_create_ml_signals.ts`](diffhunk://#diff-83f7644dc1ff87e8c5acbc265484808aa83724be5cb67ca560bafddf50c9ca3fL9-R9):
Replaced `set-value` with `@kbn/safer-lodash-set` for setting object
paths.

### Configuration updates:

*
[`renovate.json`](diffhunk://#diff-7b5c8955fc544a11b4b74eddb4115f9cc51c9cf162dbffa60d37eeed82a55a57L4554-L4572):
Removed the configuration block related to `set-value`, including
dependency grouping, reviewers, and labels.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jughosta committed:
[Discover] Improve functional test checks for breakdown (#230739)

- Closes https://github.com/elastic/kibana/issues/229703

## Summary

Had to adjust it again since
https://github.com/elastic/kibana/pull/230441 did not fully address this
flaky test.

### Checklist

- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

albertoblaz committed:
[Graph Vis] Update nodes appearance (#228403)

## Summary

Closes:
- https://github.com/elastic/kibana/issues/222436

## Changes

### New Tag component

- Renders a text badge that will map with the `entity.type` field
  - Fallbacks to "N/A" if no text is given
- Renders an optional count badge with the number of grouped entities
per node
  - Count value is abbreviated with 'k', 'm', 'b', 't'

<img width="122" height="58" alt="Screenshot 2025-07-21 at 17 31 15"
src="https://github.com/user-attachments/assets/db8840a8-90e2-4aef-8a74-b8ba402bd5f5"
/>
<br />
<img width="77" height="41" alt="Screenshot 2025-08-01 at 11 50 44"
src="https://github.com/user-attachments/assets/640ebd08-2399-4f02-8578-3b68bd53941d"
/>
<br />
<img width="143" height="49" alt="Screenshot 2025-07-21 at 17 37 42"
src="https://github.com/user-attachments/assets/9dcc3d3c-b436-4c39-aa5e-4eb223e68392"
/>
<br />
<img width="163" height="45" alt="Screenshot 2025-08-01 at 11 56 15"
src="https://github.com/user-attachments/assets/bfea57fc-3320-422a-a5b9-61d823603924"
/>

### New IPs component

- Renders a single IP address with an optional "+X" count if multiple
IPs are given
- When hovered or focused, it opens up a tooltip
  - Tooltip title is "IP Addresses"
- Max 10 IPs are listed one by one, with copy to prompt user to open
full details in flyout

<img width="239" height="53" alt="Screenshot 2025-08-01 at 11 52 29"
src="https://github.com/user-attachments/assets/df525933-dc2e-4568-9b96-db45961d9bab"
/>
<br />
<img width="400" height="212" alt="Screenshot 2025-08-01 at 11 51 23"
src="https://github.com/user-attachments/assets/c0cc24c0-1592-4229-a53a-8fec8f93c328"
/>
<br />
<img width="480" height="345" alt="Screenshot 2025-08-01 at 11 52 16"
src="https://github.com/user-attachments/assets/e4a9cf70-c274-4a2f-9cf5-f7278abf3ef0"
/>

### New CountryFlags component

- Renders max 2 emoji flags, followed by a "+X" count
- When hovered or focused, it opens up a tooltip
  - Tooltip title is "Geolocation"
- Max 10 countries are listed by one one, with copy to prompt user to
open full details in flyout

<img width="94" height="48" alt="Screenshot 2025-08-01 at 11 54 03"
src="https://github.com/user-attachments/assets/db7f15e3-ea32-462b-9425-311520873c73"
/>
<br />
<img width="370" height="128" alt="Screenshot 2025-08-01 at 11 54 31"
src="https://github.com/user-attachments/assets/e7cda437-cd80-420d-b3b5-9562c28019d3"
/>
<br />
<img width="360" height="335" alt="Screenshot 2025-08-01 at 11 55 02"
src="https://github.com/user-attachments/assets/7ebb8f55-fe09-4368-9840-988e05bf4173"
/>

### 1-line, mid-truncation in Label

- Changes Label to show ellipsis in the middle
- Display value in bold style

<img width="435" height="56" alt="Screenshot 2025-07-21 at 17 31 02"
src="https://github.com/user-attachments/assets/2285844c-54d9-463b-9a31-dccb1e2a6c6a"
/>

### Generic changes

- Shadows (`useEuiShadow('m')`) are rendered when hovered, focused or
dragging
- Focus outline is now identical to the one shown on hover

## How to test

### In Kibana

1. Add this line to your `kibana.dev.yml`:

    ```yml
    uiSettings.overrides.securitySolution:enableGraphVisualization: true
    ```

2. Then, run these 2 commands while running Kibana with `yarn start
--no-base-path`. This is for setting up the local env with data.

    ```bash
node scripts/es_archiver load
x-pack/solutions/security/test/cloud_security_posture_functional/es_archives/logs_gcp_audit
--es-url http://elastic:changeme@localhost:9200 --kibana-url
http://elastic:changeme@localhost:5601
    ```

    ```bash
node scripts/es_archiver load
x-pack/solutions/security/test/cloud_security_posture_functional/es_archives/security_alerts
--es-url http://elastic:changeme@localhost:9200 --kibana-url
http://elastic:changeme@localhost:5601
    ```

3. Finally in Kibana, go to Alerts and update the date-picker to include
data from a year ago. Then check one of the alerts details opening the
right-side flyout and find the "Graph preview" section in it.

### In Storybook

1. Run in terminal:

    ```bash
    yarn storybook cloud_security_posture_graph
    ```

2. Open [http://localhost:9001/](http://localhost:9001/).

## More details

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

No risks.

nastasha-solomon committed:
[DOCS][9.0.3, 9.0.4, 9.1.0] [Known Issue] Rule issues occur when xpack.alerting.rules.run.alerts.max is set to a value greater than 5000 (#230837)

Adds the known issue about the `xpack.alerting.rules.run.alerts.max`
setting to the Kibana known issues page and marked it as a known issue
in 9.0.3, 9.0.4, and 9.1.0. Related issue:
https://github.com/elastic/kibana/issues/230275

**Corresponding 8.x PR**: https://github.com/elastic/kibana/pull/230835


[Preview](https://github.com/elastic/kibana/pull/230837#issuecomment-3160795376)

nreese committed:
[dashboard] fix duplicate panel action hangs when dashboard has collapsed section that are closed on page load (#230842)

Closes https://github.com/elastic/kibana/issues/230818

PR updates getChildApi with logic to return `undefined` if panel is in a
collapsed section.

elastic-renovate-prod[bot] committed:
Update @testing-library (main) (#230346)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
|
[@testing-library/dom](https://redirect.github.com/testing-library/dom-testing-library)
| devDependencies | patch | [`^10.4.0` ->
`^10.4.1`](https://renovatebot.com/diffs/npm/@testing-library%2fdom/10.4.0/10.4.1)
|
|
[@testing-library/jest-dom](https://redirect.github.com/testing-library/jest-dom)
| devDependencies | patch | [`^6.6.3` ->
`^6.6.4`](https://renovatebot.com/diffs/npm/@testing-library%2fjest-dom/6.6.3/6.6.4)
|

---

### Release Notes

<details>
<summary>testing-library/dom-testing-library
(@&#8203;testing-library/dom)</summary>

###
[`v10.4.1`](https://redirect.github.com/testing-library/dom-testing-library/releases/tag/v10.4.1)

[Compare
Source](https://redirect.github.com/testing-library/dom-testing-library/compare/v10.4.0...v10.4.1)

##### Bug Fixes

- **deps:** replace chalk with picocolors
([#&#8203;1341](https://redirect.github.com/testing-library/dom-testing-library/issues/1341))
([225a3e4](https://redirect.github.com/testing-library/dom-testing-library/commit/225a3e4cfaa8f8046989d51b9051df507354b644))

</details>

<details>
<summary>testing-library/jest-dom
(@&#8203;testing-library/jest-dom)</summary>

###
[`v6.6.4`](https://redirect.github.com/testing-library/jest-dom/releases/tag/v6.6.4)

[Compare
Source](https://redirect.github.com/testing-library/jest-dom/compare/v6.6.3...v6.6.4)

##### Performance Improvements

- replace chalk with picocolors
([#&#8203;659](https://redirect.github.com/testing-library/jest-dom/issues/659))
([707e647](https://redirect.github.com/testing-library/jest-dom/commit/707e6471ae33fa2a25fab7e87be721218b5b9339))

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config help](https://elastic.slack.com/archives/C07AMD4CNUR) if that's
undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://redirect.github.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzOS4xMDcuMCIsInVwZGF0ZWRJblZlciI6IjM5LjEwNy4wIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJUZWFtOk9wZXJhdGlvbnMiLCJiYWNrcG9ydDphbGwtb3BlbiIsInJlbGVhc2Vfbm90ZTpza2lwIl19-->

Co-authored-by: elastic-renovate-prod[bot] <174716857+elastic-renovate-prod[bot]@users.noreply.github.com>
Co-authored-by: Brad White <Ikuni17@users.noreply.github.com>

maximpn committed:
[Security Solution] Wait for Fleet setup completion in Cypress before running tests (#230632)

**Relates to: https://github.com/elastic/kibana/issues/230363**
**Relates to: https://github.com/elastic/kibana/pull/230338**

## Summary

This PR makes sure Cypress e2e test run after Fleet setup is finished. This approach is similar to https://github.com/elastic/kibana/pull/230338 used in our integration tests.

Fleet setup running concurrently to the tests may cause tests flakiness. This behavior is described in https://github.com/elastic/kibana/issues/230363.

maximpn committed:
[Security Solution] Fix a failed e2e prebuilt rules upgrade RBAC test (#230557)

**Resolves: https://github.com/elastic/kibana/issues/230105**

## Summary

This PR fixes a failed Cypress test. The fix makes using the correct roles in Serverless.

## Details

Using `reader` user role isn't reliable in Serverless. This PR switched to using `t1_analyst` in Serverless.

## Flaky test runner

- ✅  [Serverless 100 runs](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9026)

talboren committed:
feat: workflows connector (#229707)

### What's in this PR?
This PR implements a new workflows connector for Kibana with
comprehensive unit test coverage, enabling users to trigger workflows
when alerts are generated.

### Workflows Connector Implementation

#### Server-side Components

- Main connector (index.ts) - Connector type definition, executor, and
adapter for rule integration
- Service layer (service.ts) - External service integration with
dependency injection
- API handlers (api.ts) - Workflow execution logic with alert filtering
- Validation (validators.ts, schema.ts) - Configuration and parameter
validation
- Utilities (utils.ts, translations.ts) - Error handling and i18n
support

#### Client-side Components

- Connector UI (workflows.tsx) - Connector type registration and
validation
- Parameters UI (workflows_params.tsx) - Workflow selection dropdown
with API integration
- Supporting files (types.ts, translations.ts) - Type definitions and UI
messages

### Quick Demo



https://github.com/user-attachments/assets/9c17eeba-1202-48b1-8ef9-ba791e2fda3c

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
[ES|QL] Update grammars (#230353)

This PR updates the ES|QL grammars (lexer and parser) to match the
latest version in Elasticsearch.

---------

Co-authored-by: Sebastian Delle Donne <sebastian.delledonne@elastic.co>

jloleysens committed:
[OAS] Promote space awareness in REST docs (#230594)

rStelmach committed:
[Oblt Onboarding] Adjust e2e onboarding tests to work in serverless environment (#229969)

dmlemeshko committed:
[ska] relocate chat serverless api & functional tests (#230527)

## Summary

Part of https://github.com/elastic/kibana-team/issues/1503

This PR is mostly about moving chat FTR tests from
`x-pack/test_serverless/` dir to `x-pack/solutions/chat/test`

Before
```
x-pack/test_serverless/
  | - functional/test_suites/chat/
  | - api_integration/test_suites/chat/
```

After: 
```
x-pack/solutions/chat/test/serverless
  | - functional/
  |  |- test_suites/
  |  |- configs/
  | - api_integration/
     |- test_suites/
     |- configs/
```

All the tests are run under the renamed FTR configs:

<img width="1716" height="200" alt="image"
src="https://github.com/user-attachments/assets/56bf8459-4b5d-4504-81a0-6ea936140c09"
/>

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

sphilipse committed:
[Search] Fix null property being read on Playground (#230729)

## Summary

This prevents Playground from throwing a useless null error when being
provided with an empty search hit.

florent-leborgne committed:
[Docs] Adds release notes for 9.1.1 (#230524)

This PR adds release notes for version the 9.1.1 patch

Closes: https://github.com/elastic/docs-content/issues/2390

adcoelho committed:
[ResponseOps][Rules] Fix flaky serverless rule details tests. (#229157)

Closes #172918
Closes #172916

## Summary

**Should these tests be owned by response-ops?**

**Should they be deleted?** - The other serverless solutions don't have
similar tests.

These tests were quite outdated. The rule details UI has changed from
the old sidepanel to a dedicated page.

In this PR:
- Updated the serverless tests to navigate the new rule details UI.
- Removed a test that is not relevant anymore in the new UI.
- Added a `retry.tryForTime` call around the previously flaky calls.
- Updated the readme because the instructions to run the tests were also
outdated.

I am starting a flaky test runner that I will link here.

jedrazb committed:
[OneChat] Support a2a protocol (#230521)

## Summary

Add support for Google's A2A protocol. Allow exposing custom agents from
onechat.

Documentation: https://a2a-protocol.org/latest/specification/

### New package `@a2a-js/sdk`

- [x] **Purpose:** Google's A2A SDK lib 
- [x] **Justification:** We want to expose agents via A2A, similarly to
how we do with MCP
- [x] **Alternatives explored:** This is the official sdk
- [x] **Existing dependencies:** no 

### New API endpoints

- `/api/chat/a2a/{agentId}.json` Agentcard endpoint
- `/api/chat/a2a/{agentId}` A2A protocol endpoint, handles JSON-RPC
payload

### Demo


https://github.com/user-attachments/assets/0a752455-b17b-4cb0-8f94-83798d7ce57d

### Test this locally 

- Follow readme of https://github.com/a2aproject/a2a-inspector and start
it locally
- Make sure kibana has LLM enabled 
- In A2A inspector link to default agent:
`http://localhost:5601/api/chat/a2a/default.json` , make sure to add
auth header

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Tiago Costa <tiago.costa@elastic.co>
Co-authored-by: Joe McElroy <joseph.mcelroy@elastic.co>

weronikaolejniczak committed:
feat(navigation): make logo clickable (#230392)

## Summary

Resolves https://github.com/elastic/eui-private/issues/394

Changes:
- The logo is now clickable as per a popular request. You can pass
`href` in `logo` prop to the `<Navigation>` component.
- Interface has changed, now logo properties are grouped under the
`logo` prop.
- `isCurrent` props are renamed to `isActive` and active management
state is handled based on the item `id`s. So, if a parent and a child
have the same `id`, they both become active. From the navigation
perspective, they are the same navigational element.
  - For correct active state management, logo requires an `id` as well.\
- All menu items are links, except for the "Menu" trigger which doesn't
navigate. Therefore, all menu items support the browser context menu
"Open in a new tab" as well as middle click and <kbd>CMD</kbd> + Click.
- All menu item labels are `11px` and have a weight of `500`, except for
the logo which has `600`.
- Navigation interfaces are exported from the package so they can be
used externally by consumers.

![Kapture 2025-08-06 at 13 32
14](https://github.com/user-attachments/assets/b7875c79-df99-4184-ac17-5c4602d3bf3a)
<img width="820" height="1028" alt="Screenshot 2025-08-06 at 12 52 38"
src="https://github.com/user-attachments/assets/c12543f4-e608-4e1d-a7df-7d911ec61fa1"
/>
<img width="832" height="1026" alt="Screenshot 2025-08-06 at 12 53 03"
src="https://github.com/user-attachments/assets/dee356fe-84c9-4459-a9cf-c997504b9cdb"
/>
<img width="826" height="973" alt="Screenshot 2025-08-06 at 13 18 51"
src="https://github.com/user-attachments/assets/937179cf-dc6f-4b3e-b331-37606362d4f7"
/>
<img width="821" height="973" alt="Screenshot 2025-08-06 at 13 19 04"
src="https://github.com/user-attachments/assets/4ac58bb8-3ce3-420c-b032-5500966e34b8"
/>


## QA

Run:
`yarn storybook shared_ux`

Navigate to `/?path=/story/chrome-navigation--default`

### Checklist

- [ ] ~Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)~
(will be tackled on [another
issue](https://github.com/elastic/eui-private/issues/371))
- [ ] ~[Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios~ (will be
tackled on [another
issue](https://github.com/elastic/eui-private/issues/377))
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

---------

Co-authored-by: Anton Dosov <anton.dosov@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

rStelmach committed:
[Observability:Streams] Improve Streams table accessibility (#225659)

closes: #224738 

## Summary 📚 

Documents and Retention columns on the Streams home page were not
focusable, so keyboard users couldn’t reach the data and screen-reader
users heard nothing.

This PR adds:
- `tabIndex` and descriptive `aria-labels` to the Documents and
Retention cells, their loading states, and relevant icons/badges.
- Focusable, labelled wrapper around the Documents histogram and a
matching `ariaLabel` on the chart canvas.
- Properly labelled ILM policy badge/link, DSL text, ∞ icon, and error
badge in the Retention column.
- Accessible column headers and a table caption.
- Removal of redundant roles that triggered axe-core


## How to test 🧪 

- Open Observability → Streams with populated data.
- Navigate the table using a keyboard
- Turn on VoiceOver and navigate through the table elements.

---------

Co-authored-by: Joe Reuter <johannes.reuter@elastic.co>
Co-authored-by: Abdul Zahid <awahab07@yahoo.com>

PhilippeOberti committed:
[Security Solution] unskip explore pagination Cypress test (#230523)

## Summary

The test was indeed flaky and easily reproduced locally. For some reason
the `Refresh` button in the global KQL bar is not updating its aria
label property from `Refresh query` to `Needs updating` when clicked.

This is the correct expected behavior


https://github.com/user-attachments/assets/72a2e170-5886-480b-ab87-29bd3c7a8e52

and this is what's happening in Cypress


https://github.com/user-attachments/assets/1cac950d-a580-4252-8b3b-119c955d92f5

To make sure that the page is correctly waiting for the refresh to have
happened completely, the solution here also adds a check to that same
`Refresh` button to verify that it is not disabled (during refresh the
button is disabled).


Flaky Test Runner:
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9023
All test passed!

https://github.com/elastic/kibana/issues/165968
https://github.com/elastic/kibana/issues/221541
https://github.com/elastic/kibana/issues/220137

markov00 committed:
[Lens] Introduce shared testing utilities for RTL (#230776)

This pull request introduces a new shared package,
`@kbn/chart-test-jest-helpers`, to provide utilities for testing Elastic
Charts components with Jest and React Testing Library. The main changes
involve extracting chart testing helpers from `expression_metric` and
centralizing them for reuse, updating tests to use the new package, and
cleaning up the old implementation.

**Introduction of shared chart testing utilities:**

* Added the new package `@kbn/chart-test-jest-helpers`, including
configuration files (`package.json`, `kibana.jsonc`, `tsconfig.json`,
`jest.config.js`) and documentation (`README.md`).
* Implemented `renderChart` and `waitForRenderComplete` helpers for
rendering charts adapting them from `chart_testing_utilities.ts` and
synchronizing test execution, plus `ResizeObserverMock` and
setup/cleanup functions to mock browser behavior required by Elastic
Charts.

**Migration of chart tests to shared helpers:**

* Updated `metric_vis.test.tsx` to use the new helpers from
`@kbn/chart-test-jest-helpers`, replacing local mock setup and render
logic with imports from the shared package.

**Cleanup and removal of legacy code:**

* Removed the `chart_testing_utilities.ts` from `expression_metric`,
which previously provided local chart testing mocks and helpers.

**Dependency management:**

* Registered the new package in the root `package.json` for local
linking and usage throughout the codebase.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

alexwizp committed:
fix: [ML] Anomaly Detection: Job selector flyout missing title from announcement (#230585)

Closes: #216797

**Description**
Dialog modal, flyout, field visible title should be announced for the
users, especially using assistive technology to know what dialog modal,
flyout opened, what field is active and what is needed to enter in it.

**Changes made**
1. Added required aria attributes
2. Header was moved our of `JobSelectorFlyoutContent`

**Screen**
<img width="1356" height="657" alt="image"
src="https://github.com/user-attachments/assets/d35ebdd4-f980-4141-983a-fc6a99800a6e"
/>

alexwizp committed:
fix: [ML] Anomaly Detection: Advanced job wizard code editors missing title from announcement (#230595)

Closes: #216767

**Description**
Dialog modal, flyout, field visible title should be announced for the
users, especially using assistive technology to know what dialog modal,
flyout opened, what field is active and what is needed to enter in it.

**Changes Made:**

1. Added the required ARIA attributes.
2. Moved the title one level up to correctly set the id.

mattkime committed:
[watcher] Allow disabling watcher in any environment (#230905)

## Summary

Previously disabling watcher was only allowed for serverless but some
clients consider it a potential security risk so its useful to be able
to disable it in any environment.

To test: 

Add `xpack.watcher.enabled: false` to your kibana.dev.yml file. 
Go look for watcher in management, it should be missing.
Change `xpack.watcher.enabled:` to true.
Go look for watcher, it should be back!

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

flash1293 committed:
🌊 Streams: Do not try to patch write streams (#230559)

Fixes https://github.com/elastic/streams-program/issues/266

Currently, streams tries to patch the write index mapping when changing
things to avoid a rollover. Only if this fails, a rollover is performed.
However, this has some downsides:
* The logic we used that strips out some of the settings that are
returned from the API but can't be set programmatically can change in
Elasticsearch, leading to bugs
* For ESQL unmapped fields, mappings in an index need to stay consistent
to avoid disappearing data
* The logic for merging mappings has subtle differences between a
`putMapping` call and a `rollover` call (too many to manage them
properly)
* When fields are deleted, `putMappings` doesn't remove mapped fields

Just always rolling over removes these problems. The downside is that
more indices might be created. We need to adjust the UI to avoid this
(e.g. allowing the user to map multiple fields at once)

---------

Co-authored-by: Milton Hultgren <milton.hultgren@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

bartoval committed:
[ES|QL] Add observability tier check for functions (#230706)

## Summary

This PR is part of https://github.com/elastic/kibana/issues/216791
- **Partial** refactoring to pass callbacks instead of plain parameters,
so we don’t have to import pass-through parameters across many modules

- Updated the script to generate observability tier in function
definitions
- Introduced observability tier verification for functions

example:

use kibana in local ( Editor does not show **CATEGORIZE** suggestion)
`yarn start` and `yarn es snapshot --license trial `

<img width="1003" height="439" alt="normal"
src="https://github.com/user-attachments/assets/b622c4c2-6cbb-420c-a59e-ee4a315124a3"
/>


use kibana servless oblt( Editor show **CATEGORIZE** suggestion)
`yarn serverless-oblt `and `yarn es serverless --projectType oblt`

<img width="1588" height="605" alt="serveless"
src="https://github.com/user-attachments/assets/fbf0d56d-6d93-446a-878c-a0a9288c9b90"
/>

delanni committed:
[CI] Remove Bazel (#193983)

## Summary
Removes Bazel from Kibana.

TODOs:
 - [x] bazel replaced for install
 - [x] bazel replaced for package build
 - [x] bazel replaced for other tasks
 - [x] bazel removed as a dependency
 - [x] bazel mentions removed
 - [x] stop using `bazel-*` directories for output
 - [x] check why bundle sizes changed
 - [x] replacements for package builds
 - [x] reinstall if package.json changed 
 - [x] merge / improve 3 implementations of shared package builds
 - [x] verify pointer-compression can still work without bazel

Notes:
- `EUI_AMSTERDAM` and `euiAmsterdam` flag handling in `kbn_pm` scripts
is no longer necessary, as we don't need to inject this to Bazel, and
env variables are passed along to spawned processes

Solves: https://github.com/elastic/kibana-operations/issues/273

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Tiago Costa <tiago.costa@elastic.co>
Co-authored-by: Robert Jaszczurek <92210485+rbrtj@users.noreply.github.com>

delanni committed:
skip failing suite (#230949)

efegurkan committed:
[FTR] Unskip synonyms tests (#228032)

## Summary

Added Retries and stabilizes Synonym tests.
Moved the tests to correct config.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.


- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

denar50 committed:
adjust copy

denar50 committed:
add cypress e2e test
